OpenSSL-Strategy.md

```diff
@@ -0,0 +1,361 @@
+
+Aside from some manual configuration that is
+required in order to support GYP builds (instead of the Perl-based Configure
+script that OpenSSL ships) as described in
+[deps/openssl/doc/UPGRADING.md](https://github.com/nodejs/node/blob/master/deps/openssl/doc/UPGRADING.md),
+there are 4 floating patches that are applied on top of the plain OpenSSL 1.0.2
+source:
+
+* [c66c3d9fa](https://github.com/nodejs/node/commit/c66c3d9fa3f5bab0bdfe363dd947136cf8a3907f):
+  `deps: fix openssl assembly error on ia32 win32`. A minor tweak to
+  deps/openssl/openssl/crypto/perlasm/x86masm.pl to switch the instruction set
+  referenced in ASM from 486 to 686, only affecting Windows on IA32.
+* [42a8de2ac](https://github.com/nodejs/node/commit/42a8de2ac66b6953cbc731fdb0b128b8019643b2):
+  `deps: fix asm build error of openssl in x86_win32`. A fix for
+  deps/openssl/openssl/crypto/perlasm/x86masm.pl for ASM produced for Windows on
+  IA32 as described in
+  https://mta.openssl.org/pipermail/openssl-dev/2015-February/000651.html
+* [2eb170874](https://github.com/nodejs/node/commit/2eb170874aa5e84e71b62caab7ac9792fd59c10f):
+  `openssl: fix keypress requirement in apps on win32`. A fix for the `openssl`
+  client application, used by the Node.js test suite, to properly accept stdin
+  without requiring a physical keyboard. As described in
+  <http://openssl.6102.n7.nabble.com/PATCH-s-client-Fix-keypress-requirement-with-redirected-input-on-Windows-td46787.html>.
+* [664a65969](https://github.com/nodejs/node/commit/664a6596960655e214fef25e74d3285097703e95):
+  `deps: add -no_rand_screen to openssl s_client`. Adds a `-no_rand_screen`
+  command line option to the `openssl` client application, used by the Node.js
+  test suite which skips invocation of the `RAND_screen()` call on Windows in
+  order to speed up some of the Node.js TLS tests. Use of this can be found in
+  many of the `test/*/test-{tls,https}-*` test files.
+
+
+### OpenSSL 1.1.0
+
+OpenSSL 1.1.0 represents a fairly major rework of the codebase, at least in
+comparison to its slow evolution until this point. The external API has some
+major departures from 1.0.2. However, support for compiling against an external
+OpenSSL 1.1.0 library (dynamically linked) was
+[added](https://github.com/nodejs/node/pull/16130) to Node.js' master branch in
+late 2017.
+
+Even though OpenSSL 1.1.0 is only supported until 2019-09-11, the API shift was
+an important stage in Node.js' adaptation.
+
+Openssl 1.1.0 is currently included in Node.js 10.x, but Node.js is expected to
+be updated to include OpenSSL 1.1.1 in the upcoming 10.16.0 release.
+
+### OpenSSL 1.1.1
+
+The OpenSSL team has designated 1.1.1 the next LTS line and have made a
+commitment that it will remain both API and ABI compatible with OpenSSL 1.1.0,
+and will be supported until 2023-09-11.  It is the first OpenSSL version to
+support TLS 1.3, however Node.js' TLS1.3 support requires at least OpenSSL
+1.1.1b.
+
+### OpenSSL 3.0.0
```

Comment:
This entire section is new, and has some bad news now that the OpenSSL project has released more information about their FIPS support plan, PTAL.